flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartition.java

```diff
@@ -196,7 +197,7 @@ public boolean addBufferConsumer(BufferConsumer bufferConsumer, int subpartition
 			subpartition = subpartitions[subpartitionIndex];
 
 			if (bufferPersister != null) {
-				bufferPersister.add(bufferConsumer.copy(), subpartitionIndex);
+				bufferPersister.add(bufferConsumer, subpartitionIndex);
```

Comment:
I guess we still need to copy the `BufferConsumer`. If one `BufferConsumer` instance is referenced by two sides (netty thread and persister writer thread), and one side builds this `BufferConsumer` firstly, then the other side can not get the proper read position when building later. So `copy` can maintain separate reader position for every side.